e2e/stack_monitoring_test.go

```diff
@@ -0,0 +1,614 @@
+		}
+
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			// Internal collection will index kibana_settings.xpack.default_admin_email as null
+			// whereas Metricbeat collection simply won't index it. So if we find kibana_settings.xpack.default_admin_email
+			// is null, we simply remove it
+			if docType == "kibana_settings" {
+				err := legacy.DeleteP("xpack.default_admin_email")
+				if err != nil {
+					return fmt.Errorf("Could not remove default_admin_email field")
+				}
+			}
+
+			return nil
+		}
+	case sm.Product == "logstash":
+		sm.Port = strconv.Itoa(9601)
+
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			pipelinesPath := "logstash_stats.pipelines"
+
+			if docType == "logstash_stats" {
+				legacyPipelines := legacy.Path(pipelinesPath)
+				metricbeatPipelines := metricbeat.Path(pipelinesPath)
+
+				legacyPipeline := legacyPipelines.Children()[0]
+				metricbeatPipeline := metricbeatPipelines.Children()[0]
+
+				legacyVertices := legacyPipeline.Path("vertices")
+				metricbeatVertices := metricbeatPipeline.Path("vertices")
+
+				// sort by vertex ID
+				log.Debug(legacyVertices)
+				log.Debug(metricbeatVertices)
+			}
+
+			return nil
+		}
+	case strings.HasSuffix(sm.Product, "beat"):
+		productIndexID = "beats"
+
+		// When Metricbeat monitors Filebeat, it encounters a different set of file IDs in
+		// `type:beats_stats` documents than when internal collection monitors Filebeat. However,
+		// we expect the _number_ of files being harvested by Filebeat in either case to match.
+		// If the numbers match we normalize the file lists in `type:beats_stats` docs collected
+		// by both methods so their parity comparison succeeds.
+		sm.handleSpecialCases = func(docType string, legacy *gabs.Container, metricbeat *gabs.Container) error {
+			filesPath := "beats_stats.metrics.filebeat.harvester.files"
+
+			if docType == "beats_stats" {
+				legacyFiles := legacy.Path(filesPath)
+				metricbeatFiles := metricbeat.Path(filesPath)
+
+				legacyFilesCount := len(legacyFiles.Children())
+				metricbeatFilesCount := len(metricbeatFiles.Children())
+
+				if legacyFilesCount != metricbeatFilesCount {
+					return fmt.Errorf("The number of harvested files in legacy (%d) and metricbeat (%d) collection is different", legacyFilesCount, metricbeatFilesCount)
+				}
+
+				log.Debugf("The number of harvested files in legacy and metricbeat collection is the same: %d", legacyFilesCount)
+			}
+
+			return nil
+		}
+
+		sm.Port = strconv.Itoa(5066)
+
+		// look up configurations under workspace's configurations directory
+		dir, _ := os.Getwd()
+		env[sm.Product+"ConfigFile"] = path.Join(dir, "configurations", "parity-testing", sm.Product+".yml")
+
+		env["serviceName"] = sm.Product
+
+		if collectionMethod == "metricbeat" {
+			env["httpEnabled"] = "true"
+			env["httpPort"] = "5066"
```

Comment:
I wonder if we should move these two under the beat case in the switch above. That's because these are only really valid when Metricbeat is monitoring a Beat. They are not valid when Metricbeat is monitoring other stack products like LS, etc.

Reply:
I think they are under the `strings.HasSuffix(sm.Product, "beat"):` scope, right?